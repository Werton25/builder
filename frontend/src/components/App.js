import FgServicesPage from "./fgServicesPage/FgServicesPage";
import {component, div} from "./elements";
import React from "react";

export default class App extends React.Component {
  render() {
    return div({id: 'app-component'},
      component(FgServicesPage)
    )
  }
}
