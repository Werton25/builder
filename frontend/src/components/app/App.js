import React from "react";
import HelpPage from "./helpPage/HelpPage";
import MainPage from "./mainPage/MainPage";
import Navbar from "./navbar/Navbar";
import {Switch, Route} from "react-router-dom";
import ReduxToastr from "react-redux-toastr";

export default class App extends React.Component {
  render() {
    return (
      <section id='app-component'>
        <Navbar/>
        <ReduxToastr/>
        <Switch>
          <Route path='/:serviceName/:methodName' component={MainPage}/>
          <Route path='/' component={HelpPage}/>
        </Switch>
      </section>
    );
  }
}
