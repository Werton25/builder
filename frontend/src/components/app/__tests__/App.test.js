import React from "react";
import renderer from "react-test-renderer";
import App from "../App";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {MemoryRouter} from "react-router-dom";

jest.mock('../helpPage/HelpPage');
jest.mock('../mainPage/MainPage');
jest.mock('../navbar/Navbar');
jest.mock('connected-react-router');
jest.mock('react-redux-toastr');

describe('App tests', () => {
  it('App renders correctly', () => {
    const store = configureStore()({});
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/']}>
            <App/>
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
