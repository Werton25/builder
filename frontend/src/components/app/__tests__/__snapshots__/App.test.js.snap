// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests App renders correctly 1`] = `
<section
  id="app-component"
/>
`;
